doc/supported_environments.rst

```diff
@@ -31,6 +31,7 @@ In addition, the package provides specialized submission templates for the follo
    supported_environments/stampede2
    supported_environments/comet
```

Comment:
> Comet will set the filesystem to read only and will not run jobs past July 15.

Because Comet will be removed from service in 3 days, I would like to remove Comet from the supported environments. This can be done in this PR or in a separate PR, but should be done in the near term -- ideally before the next release, if it comes after July 15.